drivers: can: sja1000: consolidate SFF/EFF payload loops

ndrs-pst opened the pull request:
Utilize `data_reg` to select the appropriate SFF/EFF data register base and use a single read/write loop for frame data access.
This replaces the separate SFF/EFF loops, streamlining the code and reducing its footprint.

henrikbrixandersen reviewed (changes requested):
Nice optimization, thanks. A few minor comments below:

ndrs-pst committed:
drivers: can: sja1000: consolidate SFF/EFF payload loops

Utilize `data_reg` to select the appropriate SFF/EFF data register
base and use a single read/write loop for frame data access.
This replaces the separate SFF/EFF loops, streamlining the code
and reducing its footprint.

Signed-off-by: Pisit Sawangvonganan <pisit@ndrsolution.com>
